src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -300,6 +306,8 @@ class ClusterTaskManagerTest : public ::testing::Test {
   int node_info_calls_;
   int announce_infeasible_task_calls_;
   absl::flat_hash_map<NodeID, rpc::GcsNodeInfo> node_info_;
+  std::vector<std::pair<std::function<void()>, double>> delayed_execution_;
```

Comment:
Can you make sure to call AssertNoLeaks at the end of new tests?

Can you add a test for increasing capacity on timeout?